utils/request_utils.go

```diff
@@ -25,18 +29,61 @@ func (u *RequestUtils) GetClientIpFromRequest(request *http.Request, options *co
 		for _, header := range options.ProxyHeaders {
 			ip := request.Header[header][0]
 			if len(ip) > 0 || ip != "" {
-				return ip
+				if strings.Contains(ip, ",") {
+					ips := strings.Split(ip, ",")
+					for _, extracted := range ips {
+						if ipUtils.IsValidPublicIp(strings.ReplaceAll(extracted, " ", "")) {
+							return strings.ReplaceAll(extracted, " ", "")
+						}
+					}
+				}
+				if ipUtils.IsValidPublicIp(strings.ReplaceAll(ip, " ", "")) {
+					return strings.ReplaceAll(ip, " ", "")
+				}
+			}
+		}
+		// If not public default to loopback check
+		for _, header := range options.ProxyHeaders {
+			ip := request.Header[header][0]
+			if len(ip) > 0 || ip != "" {
+				if strings.Contains(ip, ",") {
+					ips := strings.Split(ip, ",")
+					for _, extracted := range ips {
+						if ipUtils.IsLoopBack(strings.ReplaceAll(extracted, " ", "")) {
+							return strings.ReplaceAll(extracted, " ", "")
+						}
+					}
+				}
+				if ipUtils.IsLoopBack(strings.ReplaceAll(ip, " ", "")) {
+					return strings.ReplaceAll(ip, " ", "")
+				}
 			}
 		}
 	}
 
-	if ip, ok := request.Header["X-Forwarded-For"]; ok {
-		return ip[0]
+	for _, header := range ipHeaders {
+		if ips, ok := request.Header[header]; ok {
+			for _, ip := range ips {
+				if ipUtils.IsValidPublicIp(strings.ReplaceAll(ip, " ", "")) {
```

Comment:
strings.ReplaceAll(ip, " ", ""), u use in alot of places create an utility function normalizeIp/formatIp, so the code could be nices